### PR TITLE
Django thumbnails

### LINKED ISSUE #2659: Find a faster way to request signatures from S3 storage objects
### Is your feature request related to a problem? Please describe.
Staging is very slow. It takes seconds to get signatures for S3 objects. And our SSD's are getting too expensive

### Describe the solution you'd like
This should be faster, so we can move to S3.

### Motivation
Money and server load

### PR DESCRIPTION
**Pull request by JobDoesburg**
Closes #2659 

### Summary
Rewrite our complete media and thumbnailing system to Django thumbnails.

We used to have a _lazy_ thumbnailing system. That is: whenever a thumbnail was inserted, we inserted an image source with a route to a thumbnailer view that takes the path of the media to be thumbnailed as an argument. Then, only when that view was loaded, a thumbnail was retrieved (if it already exists) or created. This was a nice approach, as it keeps rendering our pages relatively fast: when loading a photo album with 50 images, this approach makes the initial page load fast, and make the thumbnails appear later.

However, when moving to an AWS S3 storage backend, this didn't seem to work anymore, as retrieving the files from S3 for each thumbnail request turns out to be pretty slow (unverified, but probably because botos3 can't parallelize the requests or something).

This PR completely switches our thumbnailing system. Thumbnails are now generated at the time they are requested. Note: this will make initial page load of an album page slower! But, hopefully, it is better compatible with S3 storage backends, so ultimately it will be faster...

Note that it is possible to pregenerate thumbnails on saving the album, as to prevent this problem!

### How to test
Thumbnails are used in:
- activemembers (MemberGroup)
- members (Profile)
- merchandise (MerchandiseItem)
- partners (multiple models)
- photos (Photo)

Furthermore, announcements (Slides) also contain images, but we never thumbnail those so I didn't implement it there

### CODE REVIEW

**Comment by JobDoesburg** [comment]
Django storages botos3 will give us signed urls for AWS, so we can (and should) get rid of private media hosting

**Comment by JobDoesburg** [comment]
Can also close #988

**Comment by JobDoesburg** [comment]
This only needs migrations to populate the thumbnails.source table, as this doesn't happen automatically by the app

**Comment by JobDoesburg** [comment]
Decided that it writing migrations for this sucks, so I implemented a fix for it in the thumbnailing app. Currently waiting for https://github.com/ui/django-thumbnails/pull/123 
Until then, we will need to write our own migrations. Let's give it some time, we can always still do it.

We also need to check our actual thumbnail sizes, and maybe rephrase small to small_300x300 to make sure we can always resize.

**Comment by JobDoesburg** [comment]
TODO:
- check the actual thumbnail sizes being used everywhere
- check the announcements app

File: website/utils/media/services.py [hunk 1]
```diff
@@ -64,24 +53,16 @@ def get_thumbnail_url(file, size, fit=True):
...
-        "name": name,
-        "thumb_path": f"thumbnails/{size_fit}/{name}",
-        "serve_path": f"thumbnails/{size_fit}/{name}",
-        "storage": f"{storage.__class__.__module__}.{storage.__class__.__name__}",
-    }
+    if isinstance(file, ThumbnailedImageFile):
+        if size == "small":
+            return get_media_url(file.thumbnails.small)
+        if size == "medium":
+            return get_media_url(file.thumbnails.medium)
```
Comment (DeD1rk) [inline website/utils/media/services.py:64]: There are other sizes defined in settings, why aren't those in here too?

Comment (DeD1rk) [reply website/utils/media/services.py]: Update: the others are used (in ThumbnailSerializers in the API), so I guess they should be used.

**Comment by ColonelPhantom** [comment]
One thing I did notice is that pages referring to invalid images now fail to load, instead of failing to load just the images.

But I guess that that's not too big of an issue.
